File: Byte&Buy.AngularSPA/src/app/core/dto/rental/user-rental-borrower-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../common/money-dto";
import { ImageThumbnailDto } from "../image/image-thumbnail";
import { RentalStatus } from "./enum/rental-status";

export interface UserRentalBorrowerResponse {
  id: Guid;
  status: RentalStatus;
  itemName: string;
  quantity: number;
  totalPricePaid: MoneyDto;
  thumbnail: ImageThumbnailDto;
  dateCreated: string;
  rentalDays: number;
  lenderName: string;
  startingRentalDate: Date; 
  endingRentalDate: Date;  
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Rental/UserRentalBorrowerResponse.cs
using ByteBuy.Core.Domain.Rentals.Enums;
using ByteBuy.Core.DTO.Public.ImageThumbnail;
using ByteBuy.Core.DTO.Public.Money;

namespace ByteBuy.Core.DTO.Public.Rental;

public sealed record UserRentalBorrowerResponse(
    Guid Id,
    RentalStatus Status,
    string ItemName,
    int Quantity,
    MoneyDto TotalPricePaid,
    ImageThumbnailDto Thumbnail,
    DateTime DateCreated,
    int RentalDays,
    string LenderName,
    DateTime StartingRentalDate,
    DateTime EndingRentalDate);